src/server/api/lib/nexmo.js

```diff
@@ -177,8 +177,9 @@ async function handleIncomingMessage(message) {
 
   const pendingMessagePart = new PendingMessagePart({
     service: 'nexmo',
-    parent_id: parentId,
-    service_message: message,
+    service_id: message['concat-ref'] || message.messageId,
+    parent_id: parentId, //do we need this anymore, now we have service_id?
```

Comment:
i'm going to leave this in for now -- i'm assuming some more bugs might come out with more testing/use, and the archaeology will be easier.  then we can clean it up.